sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -341,9 +341,11 @@ object QueryTest {
     case (a: Array[_], b: Array[_]) =>
       a.length == b.length && a.zip(b).forall { case (l, r) => compare(l, r)}
     case (a: Map[_, _], b: Map[_, _]) =>
-      val entries1 = a.iterator.toSeq.sortBy(_.toString())
-      val entries2 = b.iterator.toSeq.sortBy(_.toString())
-      compare(entries1, entries2)
+      a.size == b.size && a.keys.forall { aKey =>
+          b.keys.find(bKey => compare(aKey, bKey))
+            .map(bKey => compare(a(aKey), b(bKey)))
+            .getOrElse(false)
```

Comment:
Either way is fine but to be clear technically chaining isn't always preferred (see https://github.com/databricks/scala-style-guide#monadic-chaining). I made a PR to your branch during the review. I wonder why you picked this over the suggestion though.

Reply:
Yes, missed it, sorry. Thanks for the PR, though :+1: